REPL: folder-scoped REPLs, navigation shortcuts, and comment/timing fixes (#762)
KxSystems/kx-vscode: CHANGELOG.md

# Change log

All notable changes to the **kdb VS Code extension** are documented in this
file.

# v1.19.0

### Enhancements

- Added a **KX: Start REPL Here** context menu command in the file explorer.

>   Right-click a folder (or a file, to use its parent directory) to start a REPL

Reviewer: ```suggestion
  Right-click a folder, or a file to use its parent directory, to start a REPL
```

  rooted there. Executions always target the active REPL (the one last started

>   or focused); when none is active, they fall back to the most specific REPL

Reviewer: ```suggestion
  or focused). When no REPL is active, they fall back to the most specific REPL
```

  whose base directory contains the file, then to the workspace REPL. The base

>   directory's `mod` folder is added to the module search path (`QPATH`) so the

Reviewer: ```suggestion
  directory's `mod` folder is added to the module search path (`QPATH`) allowing the
```

>   KX modules system resolves project-local modules with ``use`…``.

Reviewer: ```suggestion
  KX modules system to resolve project-local modules with ``use`…``.
```

All notable changes to the **kdb VS Code extension** are documented in this file.

# v1.19.0

### Enhancements

> - Added a **KX: Start REPL Here** file-explorer context command that starts a REPL based in the selected folder (or a file's parent directory). Executing a file routes to the most specific REPL whose base directory contains it, falling back to the workspace REPL. The base directory's `mod` folder is added to the module search path (`QPATH`) so the KX modules system resolves project-local modules with ``use`…``.

Reviewer: ```suggestion
- Added a **KX: Start REPL Here** context menu command in the file explorer. Right-click a folder (or a file, to use its parent directory) to start a REPL rooted there. Executing a file routes to the most specific REPL whose base directory contains it, falling back to the workspace REPL. The base directory's `mod` folder is added to the module search path (`QPATH`) so the KX modules system resolves project-local modules with ``use`…``.
```

- Added REPL keyboard shortcuts:

  | Shortcut                                                              | Action                       |
  | --------------------------------------------------------------------- | ---------------------------- |
  | Ctrl+Left / Ctrl+Right (Windows, Linux), Alt+Left / Alt+Right (macOS) | Jump to previous / next word |
  | Ctrl+L                                                                | Clear the REPL               |

### Fixes

> - Fixed the REPL treating a solitary `/` as a single line rather than the start of a multiline comment when it is not closed by a `\`.

Reviewer: ```suggestion
- Fixed an issue where the REPL treated a solitary `/` as a single-line comment instead of the start of a multiline comment when it was not closed by a `\`.
```

> - Fixed the REPL mangling the repeat count of timing system commands such as `\ts:1000 exp` and `\t:100 exp`.

Reviewer: ```suggestion
- Fixed an issue where the REPL mangled the repeat count of timing system commands such as `\ts:1000 exp` and `\t:100 exp`.
```
